orchestrator_mapreduce.py

```
@@ -0,0 +1,478 @@
+                ]
+
+                if not retryable:
+                    break
+
+                log(
+                    f"🔄 MUTATE: Retrying {len(retryable)} failed subtasks "
+                    f"(attempt {retryable[0].attempts + 1}/{retryable[0].max_attempts})"
+                )
+
+                # Deduplicate protocols before mutating to avoid race conditions
+                unique_protocols = set(st.protocol for st in retryable)
```

Ruff (0.15.18)

[warning] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)